src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -53,11 +77,29 @@
             Ref("DoubleQuotedLiteralSegment"),
         ]
     ),
-    PostTableExpressionGrammar=Sequence(
-        OneOf("IGNORE", "FORCE", "USE"),
-        OneOf("INDEX", "KEY"),
-        Sequence("FOR", OneOf("JOIN"), optional=True),
-        Bracketed(Ref("ObjectReferenceSegment")),
+    FromClauseTerminatorGrammar=OneOf(
+        "WHERE",
+        "LIMIT",
+        "GROUP",
+        "ORDER",
+        "HAVING",
+        "QUALIFY",
+        "WINDOW",
+        Ref("IndexHintClauseSegment"),
+        Ref("SelectPartitionClauseSegment"),
+        Ref("ForClauseSegment"),
+        Ref("SetOperatorSegment"),
+        Ref("WithNoSchemaBindingClauseSegment"),
+    ),
+    BaseExpressionElementGrammar=OneOf(
+        Ref("LiteralGrammar"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+        Ref("IntervalExpressionSegment"),
+        Ref("ColumnReferenceSegment"),
+        Ref("ExpressionSegment"),
+        Ref("SessionVariableNameSegment"),
```

Comment:
Similar to the above, it looks like this is the same as ANSI except for these two added items. Consider using `.copy()` here as well?

Reply:
Will do, good suggestion.